library/general/src/lib/config_file/config_file.rb

```diff
@@ -0,0 +1,29 @@
+module ConfigFile
+  class ConfigFile
+    def initialize(parser, file_path, file_class: File)
+      @file_class = file_class
+      @parser = parser
+      @file_path = file_path
+    end
+
+    def write(changes_only: false)
+      merge_changes if changes_only
```

Comment:
The merging feature sounds... interesting and complicated. Too early to say more.